src/models/thread.ts

```diff
@@ -317,31 +322,33 @@ export class Thread extends ReadReceipt<EmittedEvents, EventHandlerMap> {
             this.replyCount = bundledRelationship.count;
             this._currentUserParticipated = !!bundledRelationship.current_user_participated;
 
-            const mapper = this.client.getEventMapper();
             // re-insert roomId
-            this.lastEvent = mapper({
+            this.lastEvent = new MatrixEvent({
```

Comment:
Why not using the `mapper` ? It's actually setting some event re-emitting stuff that are quite important, like dispatching `MatrixEventEvent.Decrypted`, ...

Reply:
This is something I’ve been struggling a bit with. We need those events where possible (or make sure no one uses this as if it was a full, actual event), but at the same time we need to make sure we're not leaking memory. And as techguy told me, if we just create events via the mapper but don't clean them up, we're gonna leak quite a bit of memory.